components/time/src/provider/mod.rs

```diff
@@ -69,42 +69,125 @@ impl AsULE for VariantOffsets {
     type ULE = [i8; 2];
 
     fn from_unaligned([std, dst]: Self::ULE) -> Self {
+        fn decode(encoded: i8) -> i32 {
```

Comment:
Issue: It is confusing that now you have both `VariantOffsets` and `VariantOffsetsWithFlags` that both implement AsULE. Please delete `impl AsULE for VariantOffsets`. I don't think you are using it anywhere, but if you are, then just use `VariantOffsetsWithFlags`, I think.